src/coord/src/coord.rs

```diff
@@ -264,6 +279,31 @@ impl Coordinator {
         }
     }
 
+    /// Generate a new frontiers object that forwards since changes to since_updates.
+    fn new_frontiers<I>(
+        &mut self,
+        id: GlobalId,
+        initial: I,
+        compaction_window_ms: Option<Timestamp>,
+    ) -> Frontiers<Timestamp>
+    where
+        I: IntoIterator<Item = Timestamp>,
+    {
+        let since_updates = Rc::clone(&self.since_updates);
+        let (frontier, handle) = Frontiers::new(
+            self.num_workers(),
+            initial,
+            compaction_window_ms,
+            move |frontier| {
+                since_updates.borrow_mut().insert(id, frontier);
+            },
+        );
+        let prev = self.since_handles.insert(id, handle);
+        // Ensure we don't double-register ids.
```

Comment:
If I understand correctly, we might want to throw this up on the method documentation (calling this method a second time with the same `id` will panic).

Reply:
Style-wise, fwiw, Rust doc comments often have a `# Panics` section that explains non-obvious panics. I'm not sure were there yet, as a project, but I wanted to call out the potential! :D

Reply:
Done.